### ISSUE TITLE
Build Issue with node-pre-gyp

### ISSUE THREAD

**Issue by WickedWizard3588** [TryGhost/node-sqlite3#1472]
I've been trying to compile sqlite3 for a long time now, and it keeps failing. 
NPM: 7.0.0 
Node.js: v14 LTS
OS: Ubuntu 20.04 LTS

```
ubuntu@ubuntu:~/Documents/Discord$ npm i
npm WARN deprecated node-pre-gyp@0.11.0: Please upgrade to @mapbox/node-pre-gyp: the non-scoped node-pre-gyp package is deprecated and only the @mapbox scoped package will recieve updates in the future
npm ERR! code 1
npm ERR! path /home/ubuntu/Documents/Discord/node_modules/sqlite3
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build
npm ERR! make: Entering directory '/home/ubuntu/Documents/Discord/node_modules/sqlite3/build'
npm ERR!   ACTION deps_sqlite3_gyp_action_before_build_target_unpack_sqlite_dep Release/obj/gen/sqlite-autoconf-3310100/sqlite3.c
npm ERR! make: Leaving directory '/home/ubuntu/Documents/Discord/node_modules/sqlite3/build'
npm ERR! Failed to execute '/home/ubuntu/.nvm/versions/node/v14.17.1/bin/node /home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64 --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v83' (1)
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@0.11.0
npm ERR! node-pre-gyp info using node@14.17.1 | linux | x64
npm ERR! node-pre-gyp WARN Using needle for node-pre-gyp https download 
npm ERR! node-pre-gyp info check checked for "/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64/node_sqlite3.node" (not found)
npm ERR! node-pre-gyp http GET https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v4.2.0/node-v83-linux-x64.tar.gz
npm ERR! node-pre-gyp http 403 https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v4.2.0/node-v83-linux-x64.tar.gz
npm ERR! node-pre-gyp WARN Tried to download(403): https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v4.2.0/node-v83-linux-x64.tar.gz 
npm ERR! node-pre-gyp WARN Pre-built binaries not found for sqlite3@4.2.0 and node@14.17.1 (node-v83 ABI, glibc) (falling back to source compile with node-gyp) 
npm ERR! node-pre-gyp http 403 status code downloading tarball https://mapbox-node-binary.s3.amazonaws.com/sqlite3/v4.2.0/node-v83-linux-x64.tar.gz 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.0
npm ERR! gyp info using node@14.17.1 | linux | x64
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.0
npm ERR! gyp info using node@14.17.1 | linux | x64
npm ERR! gyp info find Python using Python version 3.8.5 found at "/usr/bin/python3"
npm ERR! gyp info spawn /usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/ubuntu/Documents/Discord/node_modules/sqlite3/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/ubuntu/.cache/node-gyp/14.17.1/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/home/ubuntu/.cache/node-gyp/14.17.1',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/home/ubuntu/.cache/node-gyp/14.17.1/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/home/ubuntu/Documents/Discord/node_modules/sqlite3',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.0
npm ERR! gyp info using node@14.17.1 | linux | x64
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! /bin/sh: 1: python: not found
npm ERR! make: *** [deps/action_before_build.target.mk:13: Release/obj/gen/sqlite-autoconf-3310100/sqlite3.c] Error 127
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (events.js:375:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
npm ERR! gyp ERR! System Linux 5.8.0-43-generic
npm ERR! gyp ERR! command "/home/ubuntu/.nvm/versions/node/v14.17.1/bin/node" "/home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64/node_sqlite3.node" "--module_name=node_sqlite3" "--module_path=/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v83"
npm ERR! gyp ERR! cwd /home/ubuntu/Documents/Discord/node_modules/sqlite3
npm ERR! gyp ERR! node -v v14.17.1
npm ERR! gyp ERR! node-gyp -v v7.1.0
npm ERR! gyp ERR! not ok 
npm ERR! node-pre-gyp ERR! build error 
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/home/ubuntu/.nvm/versions/node/v14.17.1/bin/node /home/ubuntu/.nvm/versions/node/v14.17.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64/node_sqlite3.node --module_name=node_sqlite3 --module_path=/home/ubuntu/Documents/Discord/node_modules/sqlite3/lib/binding/node-v83-linux-x64 --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v83' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/ubuntu/Documents/Discord/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:375:28)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1055:16)
npm ERR! node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
npm ERR! node-pre-gyp ERR! System Linux 5.8.0-43-generic
npm ERR! node-pre-gyp ERR! command "/home/ubuntu/.nvm/versions/node/v14.17.1/bin/node" "/home/ubuntu/Documents/Discord/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
npm ERR! node-pre-gyp ERR! cwd /home/ubuntu/Documents/Discord/node_modules/sqlite3
npm ERR! node-pre-gyp ERR! node -v v14.17.1
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v0.11.0
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2021-06-20T13_18_36_733Z-debug.log
```
I've tried running `sudo apt-get install -y build-essential` reinstalling node-gyp, but it still doesn't work.
Copy of package.json
```
{
	"name": "test",
	"version": "0.1.0",
	"description": "",
	"private": true,
	"scripts": {
		"build": "rimraf dist && tsc",
		"start": "node dist/index.js",
		"deploy": "node --unhandled-rejections=strict dist/index.js",
		"lint": "eslint src --ext .ts",
		"lint:fix": "eslint src --ext .ts --fix",
		"prettier": "prettier --write **/*.{ts,js,json,yml,yaml}"
	},
	"dependencies": {
		"@keyv/sqlite": "^2.0.2",
		"discord.js": "github:discordjs/discord.js#28e961d8571276c36bc7a9743de1477e990ef585",
		"keyv": "^4.0.3",
		"luxon": "^1.26.0",
		"node-fetch": "^2.6.1",
		"winston": "^3.3.3"
	},
	"devDependencies": {
		"@commitlint/cli": "^9.1.2",
		"@commitlint/config-angular": "^9.1.2",
		"@types/keyv": "^3.1.1",
		"@types/luxon": "^1.26.2",
		"@types/node": "^14.6.4",
		"@types/node-fetch": "^2.5.8",
		"@typescript-eslint/eslint-plugin": "^3.10.1",
		"@typescript-eslint/parser": "^3.10.1",
		"eslint": "^7.7.0",
		"eslint-config-marine": "^7.2.0",
		"eslint-config-prettier": "^6.11.0",
		"eslint-plugin-prettier": "^3.1.4",
		"husky": "^4.2.5",
		"lint-staged": "^10.2.13",
		"prettier": "^2.1.1",
		"rimraf": "^3.0.2",
		"typescript": "^4.0.2"
	},
	"engines": {
		"node": ">=14.0.0"
	},
	"eslintConfig": {
		"extends": "marine/prettier/node",
		"parserOptions": {
			"project": "./tsconfig.eslint.json"
		},
		"rules": {
			"@typescript-eslint/naming-convention": 0
		}
	},
	"prettier": {
		"printWidth": 120,
		"useTabs": true,
		"singleQuote": true,
		"quoteProps": "as-needed",
		"trailingComma": "all",
		"endOfLine": "lf"
	},
	"husky": {
		"hooks": {
			"pre-commit": "lint-staged",
			"commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
		}
	},
	"lint-staged": {
		"*.js": [
			"prettier --write"
		],
		"*.ts": [
			"eslint --fix",
			"prettier --write"
		],
		"*.{json,yml,yaml}": [
			"prettier --write"
		]
	}
}
```

**Comment by Davidiusdadi** [comment]
Can confirm the issue.
The key error message is: `python: not found`
Ubuntu seems to have dropped the `python` command via (apt-)package upgrades.

My **workaround** is for now:

```
sudo apt-get install python2
sudo ln -s /usr/bin/python2 /usr/bin/python
```
This should make `npm install` / `yarn install` work again.


A **proper fix** would be to fallback to `python2` automatically if `python` does not exist

**Comment by WickedWizard3588** [comment]
I have Python 3 if that's considered a problem

**Comment by telubeer** [comment]
> I have Python 3 if that's considered a problem

You can install `python-is-python3` package this package symlinking `python3` to `python` binary

```
sudo apt-get install python-is-python3
```

**Comment by daniellockyer** [comment]
This should be fixed in `sqlite3` v5.0.3 - let me know if you still experience issues 🙂
